modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -151,6 +151,242 @@ GAPI_FLUID_KERNEL(GFluidAddW, cv::gapi::core::GAddW, false)
 
 enum Arithm { ARITHM_ABSDIFF, ARITHM_ADD, ARITHM_SUBTRACT, ARITHM_MULTIPLY, ARITHM_DIVIDE };
 
+#if CV_SIMD
+template<typename T, typename VT>
+static inline void absdiff_impl(const T in1[], const T in2[], T out[], int length, int& x)
+{
+    constexpr int nlanes = VT::nlanes;
+
+    for (; x < length; )
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            VT a = vx_load(&in1[x]);
+            VT b = vx_load(&in2[x]);
+            vx_store(&out[x], v_absdiffs(a, b));
+        }
+
+        if (x < length && length >= nlanes)
+        {
+            x = length - nlanes;
+        }
+    }
+}
+
+template<typename T>
+static inline void absdiff_simd(const T in1[], const T in2[], T out[], int length, int& x)
+{
+    if (std::is_same<T, uchar>::value)
+    {
+        absdiff_impl<uchar, v_uint8>(reinterpret_cast<const uchar*>(in1),
+                                     reinterpret_cast<const uchar*>(in2),
+                                     reinterpret_cast<uchar*>(out), length, x);
```

Comment:
IMHO casts can be moved inside `absdiff_impl` to state it once (and make the code here a bit shorter)

Reply:
This variant does not compile.